Microsoft.Toolkit.Services/Services/OneDrive/OneDriveStorageItem.cs

```diff
@@ -23,98 +26,89 @@ namespace Microsoft.Toolkit.Services.OneDrive
     /// <summary>
     /// GraphOneDriveStorageItem class.
     /// </summary>
-    public class OneDriveStorageItem
+    public class OneDriveStorageItem : INotifyPropertyChanged
```

Comment:
It feels weird extending `INotifyPropertyChanged` for just one property. This is a model level class so I don't think we should be extending the interface at all. How do you feel about removing the Thumbnail property and adding a return value to the `GetThumbnailAsync` method so the developer can create their own ViewModel?

Reply:
Let me check in that tweak just to see what it looks like, and to make sure I understand you.  (will pull it back out if I misunderstood).

Reply:
You just need to handle multiple requests (don't do them), and make sure you are on the UI thread when you set the variable back.

Reply:
commit 56649eb7 …   is that basically what you meant?

Reply:
Yes. I think this is good enough for now, but in the future we should handle multiple requests at the same time. Have you tested this? Is it working as expected? I'm afraid it might throw if the request is not on the UI thread. I would make sure it is.

Reply:
just updated again to make sure the PropertyChanged.Invoke is on the ui thread.

Reply:
66de2f2

Reply:
ok... had to tweak a condition logic bug, but that makes it behave relatively nicely.

Reply:
The project is not building. Remember that the Services project is using multi-targeting. This means that you can't use the namespace 'Windows' on it. Don't dispatch the call to the UI Thread if you are not on Windows 10. Wrap everything that uses that namespace on a #if WINRT. Also, never use async void. Change that method to "async Task<string>" (still don't await or .Wait() on it, so it still is a fire and forget method) and return the newValue variable. This makes it possible to await that operation if you are not binding to pre property, but you want it's value.

Reply:
Gotta admit, the thought that any part of this might be targeting anything other than Windows 10 never crossed my mind. Feeling a bit outta touch... sorry.